core/src/main/scala/org/http4s/parser/Rfc3986Parser.scala

```diff
@@ -14,6 +14,8 @@ private[parser] trait Rfc3986Parser { this: Parser =>
 
   def charset: Charset
 
+  def UriTotal: Rule1[org.http4s.Uri] = rule { Uri ~ EOI }
```

Comment:
This will conflict with the commit thats already in release-0.12.4: https://github.com/http4s/http4s/commit/538d3e771ff8c73b92a133a6c6e7b73e17be2f96. Do you want to wait until its merged into master (today perhaps)?